File: BatchTestTool/java/java automated tester/Test/to_be_tested/ComplexCalculator.java
package to_be_tested;

public class ComplexCalculator {

    public static Complex add(Complex c1, Complex c2) {
        double real = c1.getReal() + c2.getReal();
        double imaginary = c1.getImaginary() + c2.getImaginary();
        return new Complex(real, imaginary);
    }

    public static Complex subtract(Complex c1, Complex c2) {
        double real = c1.getReal() - c2.getReal();
        double imaginary = c1.getImaginary() - c2.getImaginary();
        return new Complex(real, imaginary);
    }

    public static complex multiply(complex c1, complex c2) {
        double real = c1.getReal() * c2.getReal() - c1.getImaginary() * c2.getImaginary();
        double imaginary = c1.getReal() * c2.getImaginary() + c1.getImaginary() * c2.getReal();
        return new complex(real, imaginary);
    }

    public static complex divide(complex c1, complex c2) {
        double denominator = c2.getReal() * c2.getReal() + c2.getImaginary() * c2.getImaginary();
        double real = (c1.getReal() * c2.getReal() + c1.getImaginary() * c2.getImaginary()) / denominator;
        double imaginary = (c1.getImaginary() * c2.getReal() - c1.getReal() * c2.getImaginary()) / denominator;
        return new complex(real, imaginary);
    }
}

public class Complex {

    public double real;

    public double imaginary;

    public Complex(double real, double imaginary) {
        this.real = real;
        this.imaginary = imaginary;
    }

    public double getReal() {
        return real;
    }

    public double getImaginary() {
        return imaginary;
    }
}

public class complex {

    public double real;

    public double imaginary;

    public complex(double real, double imaginary) {
        this.real = real;
        this.imaginary = imaginary;
    }

    public double getReal() {
        return real;
    }

    public double getImaginary() {
        return imaginary;
    }
}

File: ClassEval_T/cpp/solution/ComplexCalculator.cpp
#include <complex>

class ComplexCalculator {
public:
    static std::complex<double> add(const std::complex<double>& c1, const std::complex<double>& c2);

    static std::complex<double> subtract(const std::complex<double>& c1, const std::complex<double>& c2);

    static std::complex<double> multiply(const std::complex<double>& c1, const std::complex<double>& c2);

    static std::complex<double> divide(const std::complex<double>& c1, const std::complex<double>& c2);
};

std::complex<double> ComplexCalculator::add(const std::complex<double>& c1, const std::complex<double>& c2) {
    double real = c1.real() + c2.real();
    double imaginary = c1.imag() + c2.imag();
    return std::complex<double>(real, imaginary);
}

std::complex<double> ComplexCalculator::subtract(const std::complex<double>& c1, const std::complex<double>& c2) {
    double real = c1.real() - c2.real();
    double imaginary = c1.imag() - c2.imag();
    return std::complex<double>(real, imaginary);
}

std::complex<double> ComplexCalculator::multiply(const std::complex<double>& c1, const std::complex<double>& c2) {
    double real = c1.real() * c2.real() - c1.imag() * c2.imag();
    double imaginary = c1.real() * c2.imag() + c1.imag() * c2.real();
    return std::complex<double>(real, imaginary);
}

std::complex<double> ComplexCalculator::divide(const std::complex<double>& c1, const std::complex<double>& c2) {
    double denominator = c2.real() * c2.real() + c2.imag() * c2.imag();
    double real = (c1.real() * c2.real() + c1.imag() * c2.imag()) / denominator;
    double imaginary = (c1.imag() * c2.real() - c1.real() * c2.imag()) / denominator;
    return std::complex<double>(real, imaginary);
}
